bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusSlave.java

```diff
@@ -198,10 +200,9 @@ private void setCoil(Command command, ModbusBindingConfig config)
      * @throws ModbusException ModbusIOException on IO errors, ModbusSlaveException with protocol level exceptions
      * @throws ModbusUnexpectedTransactionIdException when response transaction id does not match the request
      */
-    protected void setRegister(Command command, ModbusBindingConfig config)
+    protected void setRegister(Command command, ModbusBindingConfig config, int readIndex, int writeRegister)

```

Comment:
Your idea is to **remove** the offset calculation _RollershutterItem.class_ of _ModbusBinding::internalReceiveCommand_

```
       if (config.getItemClass() == RollershutterItem.class) {
            if (command instanceof StopMoveType) {
                /* the second write register assigns the stop_move condition */
                slave.executeCommand(command, config, config.readIndex, config.writeIndex + 1);
                return;
            }
            if (command instanceof UpDownType) {
                /* wont read from percent read register */
                slave.executeCommand(command, config, config.readIndex + 1, config.writeIndex);
                return;
           }
```
... **remove** all the new read/write index parameters ...
and take care about the required index calculation in _ModbusSlave::setRegister_
```
        } else if (command instanceof StopMoveType) {
	    if (config.getItemClass() == RollershutterItem.class) {
	        /* the second write register assigns the stop_move condition */
		writeRegister++;
            }
            newValue = new SimpleRegister();
            if (command.equals(StopMoveType.MOVE)) {
                newValue.setValue(0);
            } else if (command.equals(StopMoveType.STOP)) {
                newValue.setValue(1);
            }
        } ...
```
and 		
```
        } else if (command instanceof UpDownType) {
            if (config.getItemClass() == RollershutterItem.class) {
	        /* wont read from percent read register */
	        readIndex++;
	    }
            newValue = readCachedRegisterValue(readIndex);
            if (newValue == null) {
                logger.warn("Not polled value for item {}. Cannot process command {}",   config.getItemName(), command);
                return;
            }...
```
 
But please keep in mind that a **new** package needs to be imported in _ModbusSlave.java_:
`import org.openhab.core.library.items.RollershutterItem;`

Please let me know whether this could be acceptable for you. Then I can change it accordingly.